scripts/tasks/unsafe-external-contributor-ci-workaround.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+
+# CI fails for external contributors because they do not have access to the secrets necessary to run
+# all jobs (namely, Sauce Labs). This script is a workaround to copy the branch from the external
+# contributor's fork and push it to the main repo. This causes CI to be executed by the internal
+# maintainer, and therefore have full access to all the secrets.
+# WARNING: This is not safe! Review all code changes in the PR before granting access to secrets.
+#
+# Usage: yarn external-contributor-ci-workaround <PR number> [remote=origin]
+
+set -e
+
+PR_INPUT="$1" # Technically can be any input `gh pr` accepts
+REMOTE="${2:-origin}"
+
+# Avoid accidentally committing unrelated files
+if [[ -n `git status --porcelain` ]]; then
+  echo -e '\033[1mPlease stash your work before continuing.\n\033[0m'
+  git status
+  exit 1
+fi
+
+# Enforce required tooling
+if [[ -z `which gh` ]]; then
+  echo 'Please install the GitHub CLI (gh): https://cli.github.com/'
+  exit 1
+fi
+
+# Enforce required input
+if [[ -z "$PR_INPUT" ]]; then
+  echo 'Please provide a pull request number or branch name.'
+  exit 1
+fi
+
+# Enforce valid remote
+if ! git remote get-url "$REMOTE" 1>/dev/null; then
+  LIST=`git remote`
+  echo "Please specify a known remote: ${LIST//$'\n'/, }"
+  exit 1
+fi
+
+# Query GitHub for PR data
+PR_DATA=`gh pr view "$PR_INPUT" --json author,baseRefName,isCrossRepository,number,title`
+AUTHOR=`echo $PR_DATA | jq -r .author.login`
```

Comment:
Not everyone has `jq` installed either; you might check for that in advance as well.